src/extra/JsonSchemaRenderer/widgets/widget-util.tsx

```diff
@@ -25,6 +26,22 @@ export interface Widget {
 	supportedTypes?: string[];
 }
 
+export const stringToNumber = function (text: string, max: number) {
```

Comment:
Why do we need to use bitwise operations, and not use `parseFloat` or `parseInt`?

Reply:
This function doesn't _parse_ the number - it basically creates a hash of a string and then limits it to a number between 0 and `max`. So I don't think `parseFloat` etc would be any use here.

Reply:
@grahammcculloch ohh I was misled by the function name and didn't really read the code thoroughly. Can we rename it to something more descriptive, `hashCode` is one naming I've seen in other languages.

Reply:
@grahammcculloch I was going to say, we have a similar function in the `jellyfish-ui-components` but it's actually the same logic (https://github.com/product-os/jellyfish-ui-components/blob/master/lib/services/helpers.js#L429-L434). I think we should abstract this to a global location and import it here and in the jf ui components

Reply:
@StefKors I'll move it to `utils/index.ts` for now. We can then always reference it in Jellyfish using `import { hashCode } from 'rendition/utils'`